src/server/bg_workers.js

```diff
@@ -209,19 +211,26 @@ function run_master_workers() {
     }
 }
 
-register_bg_worker({
-    name: 'cluster_master_publish',
-    delay: config.CLUSTER_MASTER_INTERVAL,
-    run_immediate: true
-}, cluster_master.background_worker);
+async function start_bg_workers_server() {
+    register_bg_worker({
+        name: 'cluster_master_publish',
+        delay: config.CLUSTER_MASTER_INTERVAL,
+        run_immediate: true
+    }, cluster_master.background_worker);
+
+    register_bg_worker({
+        name: 'cluster_heartbeat_writer',
+        delay: config.CLUSTER_HB_INTERVAL,
+        run_immediate: true
+    }, cluster_hb.do_heartbeat);
 
-register_bg_worker({
-    name: 'cluster_heartbeat_writer',
-    delay: config.CLUSTER_HB_INTERVAL,
-    run_immediate: true
-}, cluster_hb.do_heartbeat);
+    // Try to start the bg workers metrics server
+    prom_reporting.start_server(config.BG_METRICS_SERVER_PORT);
```

Comment:
Same comment as above. 
What happens if we fail.

Reply:
same answer